### PR TITLE
CLDC-1899 Add discounted ownership value validation

### PR DESCRIPTION
**Pull request by kosiakkatrina**
image

### CODE REVIEW

File: app/models/validations/sales/sale_information_validations.rb [hunk 1]
```diff
@@ -23,4 +23,17 @@ def validate_previous_property_unit_type(record)
       record.errors.add :fromprop, I18n.t("validations.sale_information.previous_property_type.property_type_bedsit")
     end
   end
+
+  def validate_discounted_ownership_value(record)
+    return unless record.mortgage && record.value && record.deposit && record.ownershipsch && (record.discount || record.grant)
+
```
Comment (natdeanlewissoftwire) [inline app/models/validations/sales/sale_information_validations.rb:46]: don't we also want this validation to run in the case that there is neither a discount nor a grant (i.e. `rent_to_buy_full_ownership?` == true) or is that covered by a separate validation?

**Review by natdeanlewissoftwire — APPROVED** [review]
lgtm, one query

File: app/models/validations/sales/sale_information_validations.rb (hunk 1 above)
Comment (kosiakkatrina) [reply app/models/validations/sales/sale_information_validations.rb:46]: hmm, good question! Lemme check!

Comment (kosiakkatrina) [reply app/models/validations/sales/sale_information_validations.rb:46]: Have a look now?

File: app/models/validations/sales/sale_information_validations.rb [hunk 2]
```diff
@@ -23,4 +23,20 @@ def validate_previous_property_unit_type(record)
       record.errors.add :fromprop, I18n.t("validations.sale_information.previous_property_type.property_type_bedsit")
     end
   end
+
+  def validate_discounted_ownership_value(record)
+    return unless record.value && record.deposit && record.ownershipsch
+    return unless record.mortgage || record.mortgageused == 2
```
Comment (natdeanlewissoftwire) [inline app/models/validations/sales/sale_information_validations.rb:44]: I think as part of one of Arthur's PRs mortgageuser == 2 => mortgage = 0, but I'm not sure if that behaviour was decided against in the end. Either way this should be unaffected but maybe unnecessary 👍

**Review by natdeanlewissoftwire — APPROVED** [review]

File: app/models/validations/sales/sale_information_validations.rb (hunk 2 above)
Comment (kosiakkatrina) [reply app/models/validations/sales/sale_information_validations.rb:44]: Ooh, @arfacamble do you know which change/PR that is?

Comment (arfacamble) [reply app/models/validations/sales/sale_information_validations.rb:44]: yes it's in 886
link in [PR diff](https://github.com/communitiesuk/submit-social-housing-lettings-and-sales-data/pull/1214/files#diff-140301ddbd5f381586fc3d0fbae7f21d15ef45776bcc72ce90597c5c782d07d0)
it's the work I was doing that you helped me with yesterday when I needed to move the set_derived_fields to below reset_invalidated_fields
I am waiting to finalise that PR once Nat's 1583 moves through PO review which I was expecting to happen today. (due to rebase conflict annoyances)

the upshot is that you should be able to just return unless mortgage as on line 28 and the fix making that happy will be merged in soon.

Comment (kosiakkatrina) [reply app/models/validations/sales/sale_information_validations.rb:44]: Ah nice, okay, will update it when it's merged

**Pull request merged**